### PR TITLE
Remove Routing with a Fully Qualified Type name

### PR DESCRIPTION
**Pull request by eerhardt**
This behavior is not trim-compatible, since it just a random string "route name" and calls Type.GetType with it, and then Activator.CreateInstance. The trimmer might have removed the whole type, or even the constructor on the Type. This would cause a trimmed application to behave differently than an untrimmed application.

I do not know of anyone who relies on this behavior, and there are no tests for it. So removing it seems like the best path forward.

Contributes to #1962

### CODE REVIEW

**Review by Redth — APPROVED** [review]

File: src/Controls/src/Core/Routing.cs [hunk 1]
```diff
@@ -142,23 +142,6 @@ public static Element GetOrCreateContent(string route, IServiceProvider services
 			if (s_routes.TryGetValue(route, out var content))
 				result = content.GetOrCreate(services);
 
-			if (result == null)
-			{
-				// okay maybe its a type, we'll try that just to be nice to the user
-				var type = Type.GetType(route);
```
Comment (jonathanpeppers) [inline src/Controls/src/Core/Routing.cs:148]: I don't think anyone would use a route like:
```
app://home/MyAssemblyName, MyNamespace.MyPageName
```
The space, comma, and dot would make this a very weird URL.

**Review by jonathanpeppers — APPROVED** [review]

**Comment by jonathanpeppers** [comment]
Looks like this needs to delete a test, though:
```
GlobalRouteWithDependencyResolution(Microsoft.Maui.Controls.Core.UnitTests.ShellTestBase+PageWithDependency,Microsoft.Maui.Controls.Core.UnitTests.ShellTestBase+Dependency)
```

**Review by mattleibow — APPROVED** [review]

**Comment by jsuarezruiz** [comment]
> 

Right, the build is failing because of this test.

**Review by PureWeen — APPROVED** [review]

File: src/Controls/tests/Core.UnitTests/ShellNavigatingTests.cs [hunk 2]
```diff
@@ -623,57 +623,6 @@ public async Task RouteWithGlobalPageRoute()
 			Assert.AreEqual("//animals/domestic/cats/catdetails", shell.CurrentState.Location.ToString());
 		}
 
-		[TestCase(typeof(PageWithDependency), typeof(PageWithDependency))]
```
Comment (PureWeen) [inline src/Controls/tests/Core.UnitTests/ShellNavigatingTests.cs:626]: I don't think we want to fully remove this test. Let me fix

**Review by PureWeen — CHANGES_REQUESTED** [review]

**Review by PureWeen — APPROVED** [review]

**Pull request merged**